service/plugin/src/pluginservice.cpp

```diff
@@ -37,20 +38,30 @@ void PluginServiceHandler::getPlugins(std::vector<std::string>& return_)
     return_.push_back(it.first);
 }
 
-void PluginServiceHandler::getWebPlugins(std::vector<std::string>& return_)
+void PluginServiceHandler::getThriftPlugins(std::vector<std::string>& return_)
 {
   std::string webrootDir = _configuration["webguiDir"].as<std::string>();
 
   std::vector<std::string> generatedFiles = getFiles(
     webrootDir + "/scripts/codecompass/generated");
 
-  std::vector<std::string> jsModules = getFiles(
-    webrootDir + "/scripts/codecompass/view");
-
   for (const std::string& file_ : generatedFiles)
     return_.push_back(file_.substr(webrootDir.size() + 1)); // +1 to remove starting slash
+}
+
+void PluginServiceHandler::getWebPlugins(std::vector<std::string>& return_)
+{
+  std::string webrootDir = _configuration["webguiDir"].as<std::string>();
+
+  std::vector<std::string> jsModules = getFiles(
+    webrootDir + "/scripts/release/codecompass/view");
+
+  const boost::regex pattern("^((?!\\.js\\.).)*\\.js$"); // only *.js files which does not contain the '.js.' substring
```

Comment:
If Boost.Regex supports `constexpr`, could you try:

```cpp
static constexpr boost::regex patttern(...)
```

If not, even simply doing `static const` is enough. This constructor should not be executed every time this function runs.

I would even move this to a file-scope static.

Reply:
@bruntib The release directory will contain other JS outputs, e.g `*.js.uncompressed.js` for the non-minified versions, which can be useful for debugging. However we **must** not include those.

Reply:
In this case we should rather match `.uncompressed.js` postfix. Regex is not even needed.

Reply:
There are other possible extensions, e.g. `.js.strippedConsole.js`. While we could list and match all of them (there are like 4-5 as I remember), I think this the future-proof solution.